service/src/main/kotlin/app/cash/backfila/dashboard/StopAllBackfillsAction.kt

```diff
@@ -0,0 +1,59 @@
+package app.cash.backfila.dashboard
+
+import app.cash.backfila.service.persistence.BackfilaDb
+import app.cash.backfila.service.persistence.BackfillRunQuery
+import app.cash.backfila.service.persistence.BackfillState
+import javax.inject.Inject
+import misk.MiskCaller
+import misk.hibernate.Query
+import misk.hibernate.Transacter
+import misk.hibernate.newQuery
+import misk.scope.ActionScoped
+import misk.security.authz.Authenticated
+import misk.web.Post
+import misk.web.RequestBody
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+import wisp.logging.getLogger
+
+class StopAllBackfillsRequest
+class StopAllBackfillsResponse
+
+class StopAllBackfillsAction @Inject constructor(
+  private val caller: @JvmSuppressWildcards ActionScoped<MiskCaller?>,
+  private val backfillStateToggler: BackfillStateToggler,
+  @BackfilaDb private val transacter: Transacter,
+  private val queryFactory: Query.Factory,
+) : WebAction {
+
+  @Post("/backfills/stop_all")
+  @RequestContentType(MediaTypes.APPLICATION_JSON)
+  @ResponseContentType(MediaTypes.APPLICATION_JSON)
+  @Authenticated(capabilities = ["users"])
+  fun stopAll(
+    @Suppress("unused") @RequestBody
+    request: StopAllBackfillsRequest,
+  ): StopAllBackfillsResponse {
+    // TODO check user has permissions for this service with access api
+    logger.info { "Stop all backfills called by ${caller.get()?.user}" }
+
+    val runningBackfills = transacter.transaction { session ->
```

Comment:
it's not a good idea to return Db objects out of the transaction, can you map this to return their ids?

Reply:
sure thing https://github.com/cashapp/backfila/pull/370